jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocDriverUtils.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import com.google.auto.value.AutoValue;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/** Helper methods to establish Connection for DataprocDriver. */
+public class DataprocDriverUtils {
```

Comment:
Unless these utils need to be shared by other classes, consider just making this a part of DataprocDriver. They share logic (like the jdbc prefix) so they are logically one unit. +1 to extracting the autovalue class into a separate file